app/src/ui/changes/changes-list.tsx

```diff
@@ -345,6 +347,14 @@ export class ChangesList extends React.Component<
 
     items.push(
       { type: 'separator' },
+      {
+        label: CopyFilePathLabel,
+        action: () => {
+          const fullPath = Path.join(this.props.repository.path, path)
+          clipboard.writeText(fullPath)
+        },
+        enabled: status !== AppFileStatus.Deleted,
```

Comment:
Copying the path should probably be allowed even if the file doesn't exist on disk, could be useful for people trying to look up the file on the web for example or for those who want to use it on the command line to restore it or check out a previous version.